dojo_toolkit/test_runner.py

```diff
@@ -2,59 +2,49 @@
 Module for running tests like doctest and unittest
 """
 import os
-from subprocess import PIPE, Popen, call
+import subprocess
 
 from .notifier import notifier
 
 
-class SubprocessTestRunner:
-    """
-    Base class to all test runners that use subprocess module.
-    """
-    cmd = None
-
+class DoctestTestRunner:
     def __init__(self, code_path, sound_player):
         self.code_path = code_path
         self.sound_player = sound_player
 
     def run(self):
-        """
-        run a test cmd using subprocess
-        """
-        process = Popen(
-            [self.cmd, self.code_path],
-            shell=True,
-            stdout=PIPE,
-            universal_newlines=True,
+        result = self._run_doctest()
+        self._clear_screen()
+        self._handle_result(result['is_success'])
+        print(result['output'])
+        return result['is_success']
+
+    def _run_doctest(self):
+        result = subprocess.run(
+            ["python", "-m", "doctest", self.code_path],
+            capture_output=True
         )
-        process.wait()
 
-        is_success = process.returncode == 0
-        self.handle_result(is_success)
-        command = "cls" if os.name == "nt" else "clear"
-        tmp = call(command, shell=True)  # noqa
-        print('\n'.join(str(line) for line in process.stdout.readlines()))
+        return {
+            'is_success': result.returncode == 0,
+            'output': result.stdout.decode('utf-8'),
```

Comment:
```suggestion
            'output': result.stdout,
```

Reply:
Since the `result.stdout` is now a normal utf-8 `str`, there is no more need for the `decode`